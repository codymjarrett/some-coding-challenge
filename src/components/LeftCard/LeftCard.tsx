import React from 'react'

export interface LeftCardInputProps {
	movie: any
}

const LeftCard: React.SFC<LeftCardInputProps> = ({ movie }) => (
	<div className="leftCard">
		{movie.actors.length > 0 ? (
			<div className="leftCard__content">
				<p className="detail-highlight">Actors</p>
				{movie.actors.map((actor: any, index: number) => (
					<p className="leftCard__detail" key={index}>
						{actor}
					</p>
				))}
			</div>
		) : null}
		{movie.director ? (
			<div className="leftCard__content">
				<p className="detail-highlight">Director</p>
				<p className="leftCard__detail">{movie.director}</p>
			</div>
		) : null}

		{movie.writer ? (
			<div className="leftCard__content">
				<p className="detail-highlight">
					Writer{movie.writer.split(',').length > 1 ? 's' : ''}
				</p>
				{movie.writer.split(',').map((writer: any, index: number) => (
					<p className="leftCard__detail" key={index}>
						{writer}
					</p>
				))}{' '}
			</div>
		) : null}
	</div>
)

export default LeftCard
