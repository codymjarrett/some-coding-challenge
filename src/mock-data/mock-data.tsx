export const mockData = () => {
	return {
		title: "Alexander's Ragtime Band",
		release_year: '1938',
		locations: [
			'Cliff House (1090 Point Lobos Avenue)',
			'The Walden House, Buena Vista Park',
		],
		fun_facts:
			'In 1887, the Cliff House was severely damaged when the schooner Parallel, abandoned and loaded with dynamite, ran aground on the rocks below.',
		production_company: 'Twentieth Century Fox Film Corporation',
		distributor: 'Twentieth Century Fox Film Corporation',
		director: 'Henry King',
		writer: 'Kathryn Scola',
		actor_1: 'Tyrone Power',
		actor_2: 'Alice Faye',
		actor_3: 'Don Ameche',
		actors: ['Tyrone Power', 'Alice Faye', 'Don Ameche'],
	}
}
