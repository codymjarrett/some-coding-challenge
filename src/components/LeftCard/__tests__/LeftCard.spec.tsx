import React from 'react'
import { render, cleanup } from '@testing-library/react'
import LeftCard from '../LeftCard'

import { mockData } from '../../../mock-data/mock-data'

afterEach(cleanup)

it('<LeftCard/>', () => {
	const { container } = render(<LeftCard movie={mockData()} />)
	expect(container.firstChild).toMatchSnapshot()
})
