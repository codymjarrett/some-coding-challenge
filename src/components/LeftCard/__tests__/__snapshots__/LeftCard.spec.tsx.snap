// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftCard/> 1`] = `
<div
  class="leftCard"
>
  <div
    class="leftCard__content"
  >
    <p
      class="detail-highlight"
    >
      Actors
    </p>
    <p
      class="leftCard__detail"
    >
      Tyrone Power
    </p>
    <p
      class="leftCard__detail"
    >
      Alice Faye
    </p>
    <p
      class="leftCard__detail"
    >
      Don Ameche
    </p>
  </div>
  <div
    class="leftCard__content"
  >
    <p
      class="detail-highlight"
    >
      Director
    </p>
    <p
      class="leftCard__detail"
    >
      Henry King
    </p>
  </div>
  <div
    class="leftCard__content"
  >
    <p
      class="detail-highlight"
    >
      Writer
      
    </p>
    <p
      class="leftCard__detail"
    >
      Kathryn Scola
    </p>
     
  </div>
</div>
`;
